File: src/bin/day01.py
import math

def solve(path):
    with open(path, 'r') as f:
        commands = [line.strip() for line in f if line.strip()]
    
    dial = 50
    count = 0

    for command in commands:
        print('Count ', count, ' Dial: ', dial, ' Command: ', command)
        direction = command[0]
        increment = int(command[1:])

        if direction == 'L':
            dial_before = dial
            dial -= increment
            rotations = (-dial_before + increment - 1) // 100 + 1 if dial < 0 else 0
            count += rotations
        else:
            dial_before = dial
            dial += increment
            rotations = (dial_before + increment) // 100
            count += rotations
       

        dial %= 100

    print(count)

if __name__ == "__main__":
    day01_file_path = '../../inputs/day01.txt'
    solve(day01_file_path)

File: src/days/day01.rs
use num::integer::div_floor;
use std::fs;

const DAY01_FILE_PATH: &str = "/home/pappou/Dev/aoc-2025/inputs/day01.txt";

pub fn process_command(dial: i32, command: &str) -> (i32, i32, i32) {
    let direction = command.chars().next().unwrap();
    let increment: i32 = command[1..].parse().unwrap();

    match direction {
        'L' => {
            let updated_dial = (dial - increment).rem_euclid(100);
            let count_inc_day01 = if updated_dial == 0 { 1 } else { 0 };
            let count_inc_day02 = div_floor(increment - dial + 99, 100);
            return (updated_dial, count_inc_day01, count_inc_day02);
        }
        'R' => {
            let updated_dial = (dial + increment).rem_euclid(100);
            let count_inc_day01 = if updated_dial == 0 { 1 } else { 0 };
            let count_inc_day02 = div_floor(dial + increment, 100);
            return (updated_dial, count_inc_day01, count_inc_day02);
        }
        _ => {
            panic!("Invalid command");
        }
    }
}

pub fn solve() -> (i32, i32) {
    let contents = fs::read_to_string(&DAY01_FILE_PATH).unwrap();
    let parts: Vec<&str> = contents.lines().collect();
    let mut dial: i32 = 50;
    let mut count_day01: i32 = 0;
    let mut count_day02: i32 = 0;

    for command in parts {
        let (updated_dial, updated_count_day01, updated_count_day02) =
            process_command(dial, command);
        dial = updated_dial;
        count_day01 += updated_count_day01;
        count_day02 += updated_count_day02;
    }
    (count_day01, count_day02)
}
